### PR TITLE
Support for TextMate injection grammars added.

### PR DESCRIPTION
**Pull request by dbalek**
Existing support for TextMate grammars enhanced with a possibility to register injections to the existing grammars.

### CODE REVIEW

File: ide/textmate.lexer/apichanges.xml [hunk 1]
```diff
@@ -64,6 +64,19 @@
...
+        <summary>Support for injection grammars added.</summary>
+        <version major="1" minor="3"/>
+        <date day="29" month="4" year="2019"/>
+        <author login="dbalek"/>
+        <compatibility addition="yes"/>
+        <description>
+            <p>
+                Support for injection grammars has been added.
+            </p>
+        </description>
```
Comment (JaroslavTulach) [inline ide/textmate.lexer/apichanges.xml]: Please provide a link to detailed description. I've noticed there is nice description of the usecase - try to link to it. `<a name/>` and `<a href/>` should do it.

Also list added `classes` - there seem to be some.

File: ide/textmate.lexer/arch.xml [hunk 2]
```diff
@@ -139,6 +139,33 @@
     {@literal @}GrammarRegistration(mimeType="text/x-kotlin", grammar="path/to/Kotlin.tmLanguage.json")
           </pre>
       </usecase>
+      <usecase id="register-injection-grammar" name="Register injection grammar">
+          Injection grammars let existing grammars to be extend. An injection grammar is
```
Comment (JaroslavTulach) [inline ide/textmate.lexer/arch.xml]: "to be extended"

File: ide/textmate.lexer/arch.xml [hunk 3]
```diff
@@ -139,6 +139,33 @@
...
+        &lt;/file&gt;
+    &lt;/folder&gt;
+          </pre>
+      </usecase>
+      <usecase id="register-injection-grammar-using-java-api" name="Register injection grammar using Java API">
+          To register a new injection grammar using Java API, add the GrammarInjectionRegistration
+          annotation to an appropriate package-info file, or an appropriate Java class.
+          For example:
+          <pre>
+    {@literal @}GrammarInjectionRegistration(grammar="path/to/todo-comment-injection.json", injectTo = {"source.js","source.ts"})
```
Comment (JaroslavTulach) [inline ide/textmate.lexer/arch.xml]: I don't think {@literal} works in this case. This is not a Javadoc document, but just an XML that gets converted to HTML and included inside of Javadoc files. Verify the generated Javadoc HTML page.

I'd use `<a href="@TOP/..../GrammarInjectionRegistration">` to point to the class definition.

File: ide/textmate.lexer/src/org/netbeans/modules/textmate/lexer/api/GrammarInjectionRegistration.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.textmate.lexer.api;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Target;
+
+/** Register the given TextMate grammar injection for use in the IDE.
```
Comment (JaroslavTulach) [inline ide/textmate.lexer/src/org/netbeans/modules/textmate/lexer/api/GrammarInjectionRegistration.java]: This should include the usage example or at least link to the use-case description.

File: ide/textmate.lexer/apichanges.xml (hunk 1 above)
Comment (dbalek) [reply ide/textmate.lexer/apichanges.xml]: Fixed.

File: ide/textmate.lexer/arch.xml (hunk 2 above)
Comment (dbalek) [reply ide/textmate.lexer/arch.xml]: Fixed.

File: ide/textmate.lexer/src/org/netbeans/modules/textmate/lexer/api/GrammarInjectionRegistration.java (hunk 4 above)
Comment (dbalek) [reply ide/textmate.lexer/src/org/netbeans/modules/textmate/lexer/api/GrammarInjectionRegistration.java]: Usage example added to javadoc.

File: ide/textmate.lexer/arch.xml (hunk 3 above)
Comment (dbalek) [reply ide/textmate.lexer/arch.xml]: Using the same style as the already existing `Register grammar using Java API` use-case.

**Comment by angelozerr** [comment]
@mickaelistria told me that Netbeans have integrated TextMate with TM4E, it's a super news for TM4E !

@dbalek @geertjanw @JaroslavTulach  is it possible to add this information in the TM4E README at https://github.com/eclipse/tm4e#who-is-using-tm4e by creating a PR. Thanks!

**Pull request merged**